tests/integration/install-kind.sh

```diff
@@ -59,14 +59,13 @@ function ensure_kind_network() {
   fi
 }
 
-function install_cloud_provider_kind() {
-  # https://github.com/kubernetes-sigs/cloud-provider-kind
-  if [[ ! -d cloud-provider-kind ]]; then
-    git clone --depth 1 https://github.com/kubernetes-sigs/cloud-provider-kind
-  fi
-  pushd cloud-provider-kind
-  sudo bash -c 'NET_MODE=kind docker compose up -d'
-  popd
+function run_cloud_provider_kind() {
+docker run -d \
+    --network kind \
+    --restart unless-stopped \
+    --privileged \
```

Comment:
We refer to it by name in this repo and others:
```suggestion
    --privileged \
    --name cloud-provider-kind-cloud-provider-1 \
```

Reply:
Then I think we shouldn't...or we should update them, no?
`cloud-provider-kind-cloud-provider-1` seems wrong in any case....

Reply:
We do need to be able to restart the container because it is buggy and fails to bring up load balancers for new clusters. Open to changing the name to something more appropriate.

Reply:
I guess I'm saying that even if it's called different things in different repos, they don't depend on one another directly, so although aligning the name is ideal, it shouldn't make any difference as the tests are independent. In any case, in connect, we are now using the same name as in this PR.

Reply:
Sort of independent, although I generally run the container on my machine and don't really want to recreate it with a different name when switching between projects.

Reply:
Which project has a different name at this point? Maybe we could have a smart restart script that doesn't care about the name?